homeassistant/core.py

```diff
@@ -1025,21 +1025,17 @@ def async_fire(
         listeners = self._listeners.get(event_type, [])
         match_all_listeners = self._match_all_listeners
 
+        event = Event(event_type, event_data, origin, time_fired, context)
```

Comment:
I guess we could still conditionally create this?

Not sure how much overhead this brings though...